### PR TITLE
906 Update sliding window for ROI size

### LINKED ISSUE #906: sliding_window_inference() in monai.inferers went wrong when roi_size=(M,N,1)
**Describe the bug**
sliding_window_inference() gives NaN when roi_size=(M,N,1)

**To Reproduce**
1. Go to sliding_window_inference() in monai.inferers
2. Go to the line: run
slices = dense_patch_slices( image_size=(450, 344, 450), roi_size=(96, 96, 1), scan_interval=(72,72,0))

**Expected behavior**
You will get len(slices)=30, which should be 30*450 instead.

 - MONAI latest
 - docker monai:latest

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #906 .

### Description
This PR updated the behavior for the case that `ROI * (1 - overlap)` is a float number.
Now change to `math.ceil` to cover corner case that it's smaller than 1.

### Status
**Ready**

### Types of changes

- [ ] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/inferers/utils.py [hunk 1]
```diff
@@ -173,5 +174,5 @@ def _get_scan_interval(
             scan_interval.append(int(roi_size[i]))
         else:
             # scan interval is (1-overlap)*roi_size
-            scan_interval.append(int(roi_size[i] * (1 - overlap)))
+            scan_interval.append(math.ceil(int(roi_size[i]) * (1 - overlap)))
```
Comment (wyli) [inline monai/inferers/utils.py:177]: perhaps 
```py
_interval = max(int(roi_size[i] * (1 - overlap)), 1)
scan_interval.append(_interval)
``` 
could preserve most of the previous behaviours?

Comment (Nic-Ma) [reply monai/inferers/utils.py:177]: I feel that if we only use `ceil()` for `ROI=1` and still use `floor() or int()` for `ROI>1`, the logic will be different.
@ericspod , could you please help share some comments here?

Thanks.

Comment (ericspod) [reply monai/inferers/utils.py:177]: Using max as @wyli has it makes more sense I think, with ceil the size would be 1 larger than expected much of the time, `int(roi_size[i] * (1 - overlap))` will be 0 if `overlap` is near 1 itself in which case a interval of just 1 would work. 

This function in general though needs to be documented, the arguments explained, and the asserts replaced with checks.

Comment (Nic-Ma) [reply monai/inferers/utils.py]: OK, updated in the latest commit.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli and @ericspod ,

Thanks for your review, I updated the PR according to your comments.

**Pull request merged**